packages/alfa-css/src/grammars/color.ts

```diff
@@ -101,6 +109,116 @@ function rgbaColor(stream: Stream<Token>): Color {
   return color;
 }
 
+function hslaColor(stream: Stream<Token>): Color {
+  const args = functionArguments(stream);
+
+  if (args.length !== 3 && args.length !== 4) {
+    return Transparent;
+  }
+
+  let h = 0;
+  let s = 0;
+  let l = 0;
+  let a = 1;
+
+  for (let i = 0, n = args.length; i < n; i++) {
+    const component = args[i];
+
+    switch (i) {
+      case 0:
+      case 3:
+        if (args[i].type !== TokenType.Number) {
+          return Transparent;
+        }
+        break;
+      case 1:
+      case 2:
+        if (args[i].type !== TokenType.Percentage) {
+          return Transparent;
+        }
+    }
+
+    let { value } = <Number | Percentage>component;
+
+    if (i === Component.Alpha) {
+      value = clamp(value, 0, 1);
+    } else {
+      if (component.type === TokenType.Percentage) {
+        value *= 0x64;
+      }
+
+      value = clamp(value, 0, 360);
+    }
+
+    switch (i) {
+      case Component.Red:
+        h = value;
+        break;
+      case Component.Green:
+        s = value;
+        break;
+      case Component.Blue:
+        l = value;
+        break;
+      case Component.Alpha:
+        a = value;
```

Comment:
The `Component` enum doesn't make sense for HSL(A) so just use numeric constants instead, that's fine.

Reply:
I've added the HSL constants to the components enum, as they share the same fundamental structure.